simulator/mips/mips_instr.cpp

```diff
@@ -43,8 +43,8 @@ template<typename I> auto mips_bnel    = ALU::branch<I, ALU::ne<I>>;
 template<typename I> auto mips_break   = ALU::breakpoint<I>;
 template<typename I> auto mips_clo     = ALU::clo<I, uint32>;
 template<typename I> auto mips_clz     = ALU::clz<I, uint32>;
-template<typename I> auto mips_dadd    = ALU::addition_overflow<I, uint64>;
-template<typename I> auto mips_daddi   = ALU::addition_overflow_imm<I, uint64>;
+template<typename I> auto mips_dadd    = ALU::addition_overflow_restored<I, uint64>;
+template<typename I> auto mips_daddi   = ALU::addition_overflow_imm_restored<I, uint64>;
```

Comment:
Why do we need "restored" and "not restored" functions? Only one is used if I understand correctly.

Reply:
MIPS32 ISA manual:
• If the addition results in 32-bit 2’s complement arithmetic overflow, the destination register is not modified and
an Integer Overflow exception occurs.
• If the addition does not overflow, the 32-bit result is sign-extended and placed into GPR rt.
So I made a separate function fow this behaviour.

Reply:
Yes, and why do we need a second behavior?

Reply:
I thought that for universality of alu it should be considered as a separate case. I'll remove it.